api/search.go

```diff
@@ -193,78 +173,225 @@ func CreateRequests(w http.ResponseWriter, req *http.Request, validator QueryPar
-		reqSearch.NlpCategories = nlpCriteria.Categories
+	if nlpCriteria != nil {
+		if nlpCriteria.UseCategory {
+			reqSearch.NlpCategories = nlpCriteria.Categories
+		}
+		if nlpCriteria.UseSubdivision {
+			reqSearch.NlpSubdivisionWords = nlpCriteria.SubdivisionWords
+		}
 	}
 
-	if nlpCriteria != nil && nlpCriteria.UseSubdivision {
-		reqSearch.NlpSubdivisionWords = nlpCriteria.SubdivisionWords
-	}
+	return reqSearch
+}
 
-	// population types only used if provided
+func parsePopulationTypes(params url.Values) []*query.PopulationTypeRequest {
 	popTypesParam := paramGet(params, ParamPopulationTypes, "")
-	if popTypesParam != "" {
-		popTypes := strings.Split(popTypesParam, ",")
-		p := make([]*query.PopulationTypeRequest, len(popTypes))
-		for i, popType := range popTypes {
-			p[i] = &query.PopulationTypeRequest{
-				Key: popType,
-			}
-		}
-		reqSearch.PopulationTypes = p
+	if popTypesParam == "" {
+		return nil
+	}
+	popTypes := strings.Split(popTypesParam, ",")
+	p := make([]*query.PopulationTypeRequest, len(popTypes))
+	for i, popType := range popTypes {
+		p[i] = &query.PopulationTypeRequest{Key: popType}
 	}
+	return p
+}
 
-	// dimensions only used if provided
+func parseDimensions(params url.Values) []*query.DimensionRequest {
 	dimensionsParam := paramGet(params, ParamDimensions, "")
-	if dimensionsParam != "" {
-		dims := strings.Split(dimensionsParam, ",")
-		d := make([]*query.DimensionRequest, len(dims))
-		for i, dim := range dims {
-			d[i] = &query.DimensionRequest{
-				Key: dim,
-			}
+	if dimensionsParam == "" {
+		return nil
+	}
+	dims := strings.Split(dimensionsParam, ",")
+	d := make([]*query.DimensionRequest, len(dims))
+	for i, dim := range dims {
+		d[i] = &query.DimensionRequest{Key: dim}
+	}
+	return d
+}
+
+func parseDatasetIDs(ctx context.Context, params url.Values) (datasetIDs []string, err string) {
+	datasetIDParam := paramGet(params, ParamDatasetIDs, "")
+	if datasetIDParam != "" {
+		datasetIDs = sanitiseURLParams(datasetIDParam)
+		disallowed, validationErr := validateDatasetIDs(datasetIDs)
+		if validationErr != nil {
+			log.Warn(ctx, validationErr.Error(), log.Data{"param": ParamDatasetIDs, "value": datasetIDParam, "disallowed": disallowed})
+			return nil, fmt.Sprintf("Invalid dataset_ids: %s", strings.Join(disallowed, ","))
 		}
-		reqSearch.Dimensions = d
 	}
+	return datasetIDs, ""
+}
 
+func validateDatasetIDs(datasetIDs []string) (invalidDatasetIDs []string, err error) {
+	if len(datasetIDs) == 0 {
+		return nil, nil
+	}
+
+	for _, did := range datasetIDs {
+		if did == "" {
+			invalidDatasetIDs = append(invalidDatasetIDs, "<blank>")
+		} else if len(did) <= 1 {
+			invalidDatasetIDs = append(invalidDatasetIDs, did)
+		}
+	}
+
+	if len(invalidDatasetIDs) > 0 {
+		err = fmt.Errorf("datasetID(s) not valid: %v", invalidDatasetIDs)
+	}
+
+	return invalidDatasetIDs, err
+}
+
+func parseTopics(ctx context.Context, params url.Values) (topics []string, err string) {
```

Comment:
```suggestion
func parseTopics(ctx context.Context, params url.Values) (topics []string, err error) {
```
Is there a reason for using `string`s where one would normally/consistently return `error`?

These funcs seems to bounce between the two return types for a reaons that is unclear.